### PR TITLE
Unify value methods

### PR DESCRIPTION
**Pull request by elenatorro**
Related issue https://github.com/CartoDB/carto-vl/issues/1288

- Unify `value` getter for all the expressions
- Remove `.expr` and use `value` instead
- Add `getLegendData` to base expression
- Add `values` getter (get the `data` from `getLegendData`). Also related: https://github.com/CartoDB/carto-vl/issues/1208
- Remove `eval()` when it should be `value` instead (`eval()` should be used to *evaluate* features)
- Add documentation about how to get values from expressions
- Allow users to define `othersLabel` in `top` and `buckets` expressions, and `samples` in `linear` expression, related with https://github.com/CartoDB/carto-vl/issues/1291

### CODE REVIEW

File: docs/guides/03-style-with-expressions.md [hunk 1]
```diff
@@ -99,13 +99,85 @@ To access feature properties from a dataset, use the dollar sign (`$`). For exam
...
+Once a visualization has been created, you can access and modify all styling properties and variables through the returned `Visualization` object.
 
 Once you have a `Visualization` object, you can:
-- Get its property values by using the `.value` getter (`const currentColor = viz.color.value;`)
-- Modify its property values by using the `.blendTo()` method (`viz.color.blendTo(‘green’);`)
+
+
+#### Access
+
+There three main use cases:
```
Comment (VictorVelarde) [inline docs/guides/03-style-with-expressions.md]:
```suggestion
There are three main use cases:
```

File: src/renderer/viz/expressions/RampImage.js [hunk 2]
```diff
@@ -43,9 +47,8 @@ export default class RampImage extends Base {
     }
 
     getLegendData (options) {
-        const config = Object.assign({}, DEFAULT_OPTIONS, options);
         const type = this.input.type;
-        const legendData = this.input.getLegendData(config);
+        const legendData = this.input.getLegendData(options);
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/RampImage.js]: Why have we omitted here the previous DEFAULT_OPTIONS?

File: src/renderer/viz/expressions/Zoomrange.js [hunk 3]
```diff
@@ -26,24 +26,25 @@ import { implicitCast, checkType, checkExpression } from './utils';
  * @api
  */
 export default class Zoomrange extends BaseExpression {
-    constructor (zoomBreakpointList) {
-        zoomBreakpointList = implicitCast(zoomBreakpointList);
+    constructor (_zoomBreakpointList) {
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/Zoomrange.js]: Shouldn't the ctor param be named without underscore?

File: src/renderer/viz/expressions/base.js [hunk 4]
```diff
@@ -65,19 +65,30 @@ export default class Base {
...
+
+    /**
+     * Get the expression values
+     *
+     * @api
+     * @memberof Expression
+     * @name values
+     */
+    get values () {
+        return this.getLegendData().data;
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/base.js]: Does this always guarantee a result? Should we consider a runtime error + a proper message?

File: src/renderer/viz/expressions/basic/constant.js [hunk 5]
```diff
@@ -29,17 +29,24 @@ export default class Constant extends BaseExpression {
         checkNumber('constant', 'x', 0, x);
 
         super({});
-        this.expr = x;
+        this.value = x;
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/basic/constant.js]: Are we missing here the underscore?
```suggestion
        this._value = x;
```

File: src/renderer/viz/expressions/linear.js [hunk 6]
```diff
@@ -3,7 +3,8 @@ import { checkExpression, implicitCast, checkType, checkMaxArguments, clamp } fr
 import { globalMin, globalMax } from '../expressions';
 import { castTimeRange, msToDate } from '../../../utils/util';
 import IdentityCodec from '../../../codecs/Identity';
-import TZDate from '../../../utils/time/TZDate';
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/linear.js]: I find the new name much clearer, but I guess that we should consider this a breaking change

File: src/renderer/viz/expressions/linear.js [hunk 7]
```diff
@@ -30,7 +31,7 @@ import TZDate from '../../../utils/time/TZDate';
 * @api
 */
 export default class Linear extends BaseExpression {
-    constructor (input, min, max, range) {
+    constructor (input, min, max, range, samples = DEFAULT_SAMPLES) {
```
Comment (VictorVelarde) [inline src/renderer/viz/expressions/linear.js]: What is the use case for introducing here this optional param?

File: src/renderer/viz/expressions/linear.js (hunk 6 above)
Comment (elenatorro) [reply src/renderer/viz/expressions/linear.js]: It's not an expression, why should it be then a breaking change?

File: src/renderer/viz/expressions/RampImage.js (hunk 2 above)
Comment (elenatorro) [reply src/renderer/viz/expressions/RampImage.js]: By default, the default options (`othersLabel` and `samples`) are the same, but they're defined in the constructor of the nested expression. I forgot to added in the PR description (now it's added), but this comes from this issue: https://github.com/CartoDB/carto-vl/issues/1291.

File: src/renderer/viz/expressions/Zoomrange.js (hunk 3 above)
Comment (elenatorro) [reply src/renderer/viz/expressions/Zoomrange.js]: You're right, this is a mistake when replacing the string.

File: src/renderer/viz/expressions/linear.js (hunk 7 above)
Comment (elenatorro) [reply src/renderer/viz/expressions/linear.js]: Same as here: https://github.com/CartoDB/carto-vl/pull/1323#discussion_r275337504

File: src/renderer/viz/expressions/linear.js (hunk 6 above)
Comment (VictorVelarde) [reply src/renderer/viz/expressions/linear.js]: All right then, I thought it was exposed somehow.

**Comment by VictorVelarde** [comment]
Overall, it LGTM but I would feel more comfortable if we would fix the e2e docker problem first and bring it here, before we merge this in

**Comment by rjimenezda** [comment]
Some (undocumented) part of the bridge relies on the fact that getLegendData was only present on certain expressions, but I can refactor it to use something else, like check the actual return value now, or the expression type.

**Pull request merged**